app/javascript/components/cloud-volume-form/cloud-volume-form.schema.js

```diff
@@ -12,12 +12,48 @@ const changeValue = (value, loadSchema, emptySchema) => {
 
 const storageManagers = (supports) => API.get(`/api/providers?expand=resources&attributes=id,name,${supports}&filter[]=${supports}=true`)
   .then(({ resources }) => {
-    let storageManagersOptions = [];
-    storageManagersOptions = resources.map(({ id, name }) => ({ label: name, value: id }));
+    let storageManagersOptions = resources.map(({ id, name }) => ({ label: name, value: id }));
     storageManagersOptions.unshift({ label: `<${__('Choose')}>`, value: '-1' });
     return storageManagersOptions;
   });
 
+// storage manager functions to filter by capabilities:
+const equalsUnsorted = (arr1, arr2) => {
+  if (arr1.length !== arr2.length) return false;
+  const uniqueValues = new Set([...arr1, ...arr2]);
+  for (const v of uniqueValues) {
+    const aCount = arr1.filter(e => e === v).length;
+    const bCount = arr2.filter(e => e === v).length;
+    if (aCount !== bCount) return false;
+  }
+  return true;
+}
```

Comment:
is it possible to avoid for loops here?

from what I understand, this function is trying to find for duplicate items from 2 array's

`[...arr1, ...brr2].filter((item, index) => d.indexOf(item) !== index) `will return the duplicate items

Reply:
this function is for checking equality between 2 unsorted arrays.
we only want to return true if `arr1` has exactly the same items (capabilities) as `arr2`, regardless of item order.
so merely finding the dups isn't enough, we want to verify that all items in both arrays are dups.
Is there a way to do that without the `for` loop?

Reply:
could you check if this works?

`arr2.every(arr2Item => arr1.includes(arr2Item))`

Reply:
u mean something like this?
```
return arr2.every(arr2Item => arr1.includes(arr2Item)) && arr1.every(arr1Item => arr2.includes(arr1Item))
```

Reply:
so this works well:
`return arr1.length !== arr2.length && arr2.every(arr2Item => arr1.includes(arr2Item)) && arr1.every(arr1Item => arr2.includes(arr1Item));`